src/autocomplete/EmojiProvider.tsx

```diff
@@ -94,11 +101,28 @@ export default class EmojiProvider extends AutocompleteProvider {
             // For removing punctuation
             shouldMatchWordsOnly: true,
         });
+        this.recentlyUsed = Array.from(new Set(recent.get().map(getEmojiFromUnicode).filter(Boolean)));
+    }
 
-        this.recentlyUsed = Array.from(new Set(filterBoolean(recent.get().map(getEmojiFromUnicode))));
+    private async decryptEmotes(emotes: Object, roomId: string) {
+        const decryptede=new Map<string, string>();
+        const client = MatrixClientPeg.get();
+        let durl = "";
+        const isEnc=client.isRoomEncrypted(roomId);
+        for (const shortcode in emotes) {
+            if (isEnc) {
+                const blob = await decryptFile(emotes[shortcode]);
+                durl = URL.createObjectURL(blob);
+            } else {
+                durl = mediaFromMxc(emotes[shortcode]).srcHttp;
+            }
+            decryptede[shortcode] = "<img class='mx_Emote' title=':" + shortcode +
+                ":'src='" + durl + "'/>";
```

Comment:
This is vulnerable to XSS, you're just gluing strings together to form html, if `shortcode` was `"><script>blah</script>` then it'd inject a script tag. You should just pass around ReactNodes so you don't need to dangerously set inner HTML.

Reply:
done